x-pack/plugins/apm/public/components/app/service_overview/service_overview_errors_table/index.tsx

```diff
@@ -148,52 +68,75 @@ export function ServiceOverviewErrorsTable({ serviceName }: Props) {
       }
 
       return callApmApi({
-        endpoint: 'GET /api/apm/services/{serviceName}/error_groups',
+        endpoint:
+          'GET /api/apm/services/{serviceName}/error_groups/primary_statistics',
         params: {
           path: { serviceName },
           query: {
             start,
             end,
             uiFilters: JSON.stringify(uiFilters),
-            size: PAGE_SIZE,
-            numBuckets: 20,
-            pageIndex: tableOptions.pageIndex,
-            sortField: tableOptions.sort.field,
-            sortDirection: tableOptions.sort.direction,
             transactionType,
           },
         },
       }).then((response) => {
         return {
+          requestId: uuid(),
           items: response.error_groups,
-          totalItemCount: response.total_error_groups,
-          tableOptions: {
-            pageIndex: tableOptions.pageIndex,
-            sort: {
-              field: tableOptions.sort.field,
-              direction: tableOptions.sort.direction,
-            },
-          },
         };
       });
     },
-    [
-      start,
-      end,
-      serviceName,
-      uiFilters,
-      tableOptions.pageIndex,
-      tableOptions.sort.field,
-      tableOptions.sort.direction,
-      transactionType,
-    ]
+    [start, end, serviceName, uiFilters, transactionType]
   );
 
-  const {
+  const { items, requestId } = data;
+  const currentPageErrorGroups = orderBy(
     items,
-    totalItemCount,
-    tableOptions: { pageIndex, sort },
-  } = data;
+    (group) => {
+      if (sort.field === 'occurrences') {
+        return group.occurrences.value;
+      }
+      return group[sort.field];
+    },
+    sort.direction
+  ).slice(pageIndex * PAGE_SIZE, (pageIndex + 1) * PAGE_SIZE);
+
+  const groupIds = JSON.stringify(
+    currentPageErrorGroups.map(({ group_id: groupId }) => groupId).sort()
+  );
+  const {
+    data: errorGroupComparisonStatistics,
+    status: errorGroupComparisonStatisticsStatus,
+  } = useFetcher(
+    (callApmApi) => {
+      if (currentPageErrorGroups.length && start && end && transactionType) {
+        return callApmApi({
+          endpoint:
+            'GET /api/apm/services/{serviceName}/error_groups/comparison_statistics',
+          params: {
+            path: { serviceName },
+            query: {
+              start,
+              end,
+              uiFilters: JSON.stringify(uiFilters),
+              numBuckets: 20,
+              transactionType,
+              groupIds,
+            },
+          },
+        });
+      }
+    },
+    // only fetches agg results when requestId changes or group ids change
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [requestId, groupIds],
```

Comment:
Do we actually need `requestId`? Wouldn't the following achieve the same?
```ts
if (status  === FETCH_STATUS.SUCCESS && currentPageErrorGroups.length && start && end && transactionType) {
 // ...
}

Reply:
I made some tests and that indeed seems to work the same way as with the requested. Thanks for the suggestion.

Reply:
Are you sure that it works the same?

Reply:
Yeah, as I said, I made some tests and it looks like it works as it should. I believe using both `status === success` and `preservePreviousData: false` gives the same result as having the requested.

Reply:
The reason why requestId is there is because the transaction groups alone do not provide you with enough information to decide whether to refetch. E.g., if you add a noop filter in the query bar, the comparison statistics should be refetched. I guess adding `status` as a dependency here means that as soon as the first request goes into loading, this one is invalidated (and then again when it switches to success). But that sounds very coincidental to me and using requestId for explicit invalidation is IMHO a better approach.

Reply:
And what happens when the primary stats request is cached?

Reply:
If you feel like this is too complicated you can also consider calling setState() in the first hook (so primary statistics are a hook, comparison statistics use useFetcher). I'm not sure if that will clear things up, I haven't tried it out.

Reply:
@dgieselaar your point is valid, having the `requestId` makes it more clear when the comparison data is gonna be invalidated and refetched. But I also think using `status` is valid, and is also clear, even though the comparison dependencies is invalidated when the status changes to `loading`, nothing happens until it is `success`.

IMHO both implementations are fine, we just have to come up with an agreement.

cc: @sqren

Reply:
I'm not sure if both implementations are equivalent. React batches updates AFAIK. If useFetcher() calls setState(), and then awaits a promise that immediately resolves, and calls setState() again, does React render the component twice, once for each setState() call, or does it batch the two updates? If the latter is the case, this implementation now has a bug where it won't fetch or apply new comparison data if the primary request was cached (and the transaction groups on the page are the same).

Reply:
That's a valid point @dgieselaar, thanks for that, I've made some tests and it indeed wouldn't call the comparison API if the API is cached. Since the promise is immediately resolved the first setState is ignored (the one that changes the status to loading), thus the dependencies are not invalidated. I'm going to add the requestId.

Reply:
Thanks for the deep dive! Using requestId sounds like the better solution 👍